### PR TITLE
[NVIDIA] CUDA Graph API C++ wrappers

### PR DESCRIPTION
**Pull request by ytorzuk-altran**
### Details:
 - *cudaGraph_t C++ wrapper*
 - *cudaGraphExec_t C++ wrapper*
 - *GraphCapture C++ wrapper*
 - *GraphCaptureScope C++ wrapper*
 - *New CUDA::Event constructor which allows events creation with flags*
 - *Tests which cover CUDA Graph API C++ wrappers functionality*

### Tickets:
 - *[CVS-108536](https://jira.devtools.intel.com/browse/CVS-108536)*

### CODE REVIEW

File: modules/nvidia_plugin/src/cuda/graph.cpp
Comment (ytorzuk-altran) [reply modules/nvidia_plugin/src/cuda/graph.cpp:46]: > Use `ov::Exception` here

This exception type was choosen to keep consistency with other NVIDIA plugin components which throw `InferenceEngine::GeneralError` exception https://github.com/openvinotoolkit/openvino_contrib/blob/master/modules/nvidia_plugin/src/error.cpp#LL23C5-L23C19. I agree that exception type should be updated with a new one, but across all NVIDIA Plugin components. I'd propose to update exception type for all NVIDIA plugin components in a separate PR, but keep `InferenceEngine::GeneralError` in this PR for consitency. Otherwise it implies revising all NVIDIA Plugin codebase and updating exception types which is out of scope of this PR.

Comment (nkogteva) [reply modules/nvidia_plugin/src/cuda/graph.cpp:46]: @ytorzuk-altran you can use `throwIEException`, it is content modified in https://github.com/openvinotoolkit/openvino_contrib/pull/656/

File: modules/nvidia_plugin/tests/unit/cuda/wrappers/graph.cpp
Comment (ytorzuk-altran) [reply modules/nvidia_plugin/tests/unit/cuda/wrappers/graph.cpp:10]: ie_common.h contains declaration of [InferenceEngine::GeneralError](https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/ie/ie_common.h#L346) which is used in [CudaGraphCaptureCppWrappersTest.EventUpdate](https://github.com/ytorzuk-altran/openvino_contrib/blob/feature/cuda-graph-cpp-wrappers/modules/nvidia_plugin/tests/unit/cuda/wrappers/graph.cpp#L132) test assumption.

File: modules/nvidia_plugin/src/cuda/graph.cpp
Comment (ytorzuk-altran) [reply modules/nvidia_plugin/src/cuda/graph.cpp:46]: `throwIEException` adds additional details about exception location which previously added in the caught exception. `throwIEException` is not used here to avoid exception location details duplication.

**Review by nkogteva — APPROVED** [review]

**Comment by nkogteva** [comment]
@ytorzuk-altran are you waiting for any other reviews or we can merge it?

**Comment by ytorzuk-altran** [comment]
> @ytorzuk-altran are you waiting for any other reviews or we can merge it?

@nkogteva This PR can be merged if you consider all comments are considered resolved

**Review by apavliuk-altran — APPROVED** [review]

**Pull request merged**
